## Homework 1
*Author: Puri Rudick*

1.	Install Python (if you don’t have it already) and install NLTK.  

2.	Follow the instructions in chapter 1 of Bird-Klein for implementing a “lexical diversity” scoring routine.

3.	Go to http://www.gutenberg.org/ebooks/bookshelf/215  and select texts of different grade levels (e.g., fourth reader, fifth reader et al)
Report the lexical diversity score of each. Explain whether the result was surprising.

4.	Also compare the vocabulary size of the same three texts. Explain whether the result was surprising.  

5.	Write a paragraph arguing whether vocabulary size and lexical diversity in combination could be a better measure of text difficulty (or reading level) than either measure is by itself.


In [2]:
import nltk
from urllib import request

In [29]:
# Read in the txt files
books = {
    '''The Child's World: Third Reader by Browne, Tate, and Withers''' : 'https://www.gutenberg.org/cache/epub/15170/pg15170.txt',
    'Fourth Reader: The Alexandra Readers by Dearness, McIntyre, and Saul' : 'https://www.gutenberg.org/files/51975/51975-0.txt',
    'School Reading By Grades: Fifth Year by James Baldwin' : 'https://www.gutenberg.org/files/51000/51000-0.txt',
}

In [35]:
# Lexical_diversity function from nltk book
def lexical_diversity(text):
    return len(set(text)) / len(text)

# Print each book vocabbulary size and lexical diversity
for i in books:
    response = request.urlopen(books[i])
    text = response.read().decode('utf8')
    vocab = set(text)
    vocab_size = len(vocab)
    lexi = lexical_diversity(text)
    print(i, '\n\tVocabulary Size:', vocab_size)
    print('\tLexical Diversity: ', lexi)

The Child's World: Third Reader by Browne, Tate, and Withers 
	Vocabulary Size: 89
	Lexical Diversity:  0.00041428491630513713
Fourth Reader: The Alexandra Readers by Dearness, McIntyre, and Saul 
	Vocabulary Size: 100
	Lexical Diversity:  0.0001870620409965169
School Reading By Grades: Fifth Year by James Baldwin 
	Vocabulary Size: 123
	Lexical Diversity:  0.0004523089821944708


### Discussion

The 3 books I used in this experiment are for 3rd, 4th, and 5th grade readers.  They are from 3 different authors, also different genes. From the result above I found that the books' vocabulary size increases with grade level; 89 for 3rd grade, 100 for 4th grade, and 123 for 5th grade. In my opinion, this is expeted as older children know more words and an read longer than younger children.

The lexical diversity for each book had been analized. The exical diversity measures the number of unique words to the total size of the text. Even though vocabulary is increasing, the mix of unique words does not so the text is not confusing to the student. According to the result above, 5th grade has the higest lexical diversity score at .00045, follows by 3th grade score at .00041, and the least score is from 4th grade at .00019. It seems like even though the vocabulary size increases, the mix of unique words may or may not increase. However, the book for 5th grade still be the most complicated among the three.

From this experiment, the higher age with reading level tends to have a higher vocabulary size and a higher lexical diverity.